### PR TITLE
Restore "show all key bindings" shortcut 

### LINKED ISSUE #4625: "Ctrl + Alt + / show all key bindings" doesn't work
## 🐛 Bug

Hi all,

I just noticed that Ctrl+Alt+/  show all key bindings doesn't work. I also cannot find a menu where the key bindings are listed.

image

## To Reproduce

1. Open napari
2. Hit Ctrl+Alt+/

## Expected behavior

The shortcut should work. Also a menu for showing key bindings would be great - especially for the people who prefer having menus for things.

## Environment

 - Please copy and paste the information at napari info option in help menubar here:

```
napari: 0.4.16
Platform: Windows-10-10.0.19044-SP0
Python: 3.9.7 (default, Sep 16 2021, 16:59:28) [MSC v.1916 64 bit (AMD64)]
Qt: 5.12.9
PyQt5: 5.12.3
NumPy: 1.22.4
SciPy: 1.8.0
Dask: 0+unknown
VisPy: 0.10.0

OpenGL:
- GL version: 4.6.0 NVIDIA 511.69
- MAX_TEXTURE_SIZE: 32768

Screens:
- screen 1: resolution 1920x1200, scale 1.0

Plugins:
- PlatyMatch: 0.0.3
- RedLionfish: 0.6
- StarDist: 2021.10.25
- animation: 0.0.2
- cellpose-napari: 0.1.4
- clEsperanto: 0.18.1
- console: 0.0.4
- devbio_napari: 0.5.5
- napari-3d-ortho-viewer: 0.0.1
- napari-accelerated-pixel-and-object-classification: 0.7.4
- napari-assistant: 0.3.7
- napari-brightness-contrast: 0.1.7
- napari-clusters-plotter: 0.5.0
- napari-crop: 0.1.5
- napari-cupy-image-processing: 0.2.0
- napari-curtain: 0.1.1
- napari-folder-browser: 0.1.3
- napari-itk-io: 0.1.0
- napari-layer-details-display: 0.1.3
- napari-mouse-controls: 0.1.3
- napari-ome-zarr: 0.4.0
- napari-plot-profile: 0.2.1
- napari-plugin-search: 0.1.3
- napari-process-points-and-surfaces: 0.2.0
- napari-segment-blobs-and-things-with-membranes: 0.2.19
- napari-simpleitk-image-processing: 0.3.1
- napari-stress: 0.0.14
- napari-svg: 0.1.6
- napari-tabu: 0.1.5
- napari-time-slicer: 0.4.6
- napari-tools-menu: 0.1.15
- napari-workflow-inspector: 0.2.2
- napari-workflow-optimizer: 0.1.4
- napari_skimage_regionprops1: 0.5.2
- napari_skimage_regionprops2: 0.5.2
- ome-types: 0.2.10
- scikit-image: 0.4.16
- the-segmentation-game: 0.1.0
```

 - Any other relevant information:

I'm working on Windows 10 with a US keyboard layout.

### PR DESCRIPTION
**Pull request by Czaki**
# Description

This PR restore "show all key bindings" shortcut and adds an option to edit it from preferences dialog

## Type of change

- [x] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

fix  #4625

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change
- [ ] example: I check if my changes works with both PySide and PyQt backends
      as there are small differences between the two Qt bindings.  

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

File: napari/utils/interactions.py [hunk 1]
```diff
@@ -236,7 +236,7 @@ def hello_world(layer, event):
 }
 
 
-joinchar = '-'
+joinchar = '+'
```
Comment (Czaki) [inline napari/utils/interactions.py:239]: To be consistent with Qt shortcut representation

**Review by DragaDoncila — COMMENTED** [review]
@Czaki I just pulled this down and it works great on my Mac but on Ubuntu nothing happens when I press `Ctrl+Alt+/`. The option to change the shortcut does appear in preferences, but even if I assign a new shortcut, the preference dialog doesn't open.

**Comment by Czaki** [comment]
> on Ubuntu nothing happens when I press `Ctrl+Alt+/`

Strange. I develop this on ubuntu. Did you test it on napari plugi or it is possible that some plugin overwritten it?

**Comment by DragaDoncila** [comment]
> Did you test it on napari plugi or it is possible that some plugin overwritten it?

I tested in a brand new conda env so I don't think it's a plugin overwrite - especially since even after I changed the shortcut it still didn't work. I tried other shortcuts and they did work. It could just be something with my machine though so it would be good if someone else could try on Ubuntu.

**Comment by DragaDoncila** [comment]
This is my `napari --info`:

```
napari: 0.4.16rc2.dev79+gddafd39c
Platform: Linux-5.4.0-113-generic-x86_64-with-glibc2.31
System: Ubuntu 20.04.4 LTS
Python: 3.9.12 (main, Jun  1 2022, 11:38:51)  [GCC 7.5.0]
Qt: 5.15.2
PyQt5: 5.15.6
NumPy: 1.22.4
SciPy: 1.8.1
Dask: 2022.05.2
VisPy: 0.10.0

OpenGL:
  - GL version:  4.6 (Compatibility Profile) Mesa 21.2.6
  - MAX_TEXTURE_SIZE: 16384

Screens:
  - screen 1: resolution 1920x1080, scale 1.0
  - screen 2: resolution 1920x515, scale 1.0

Plugins:
  - console: 0.0.4
  - napari-svg: 0.1.6
  - scikit-image: 0.4.16rc2.dev79+gddafd39c
```

**Comment by Czaki** [comment]
in the console:

```python
from napari.utils.action_manager import action_manager
action_manager.trigger("napari:show_shortcuts")
```

did this show shortcuts in preferences? 

And what desktop environment do you have on ubuntu? Gnome?

**Comment by DragaDoncila** [comment]
If I run the above in console or ipython I get:

```python
>>> from napari.utils.action_manager import action_manager
>>> action_manager.trigger("napari:show_shortcuts")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/draga/napari/napari/utils/action_manager.py", line 376, in trigger
    return self._actions[name].injected()
KeyError: 'napari:show_shortcuts'
```

I am indeed using gnome.

**Comment by Czaki** [comment]
Hm. 
I think that we found the core of the problem. If the action manager does not know the action then we have a problem 

what happens if you call
```
$ pytest napari/_qt/menus/_tests/test_file_menu.py
```

did it collect 2 rest and both pass?

**Comment by DragaDoncila** [comment]
@Czaki yeah oddly enough, when I run pytest on that file in that environment, the tests pass fine...

```
(napari-edit) draga@DragaZenBook:~/napari$ python -m pytest napari/_qt/menus/_tests/test_file_menu.py
============================================================ test session starts =============================================================
platform linux -- Python 3.9.12, pytest-7.1.2, pluggy-1.0.0
PyQt5 5.15.6 -- Qt runtime 5.15.2 -- Qt compiled 5.15.2
rootdir: /home/draga/napari, configfile: pyproject.toml
plugins: napari-0.4.16rc2.dev79+gddafd39c, qt-4.0.2, hypothesis-6.47.0, napari-plugin-engine-0.2.0
collected 2 items                                                                                                                            

napari/_qt/menus/_tests/test_file_menu.py ..                                                                                           [100%]

============================================================ slowest 10 durations ============================================================
0.89s call     napari/_qt/menus/_tests/test_file_menu.py::test_sample_data_triggers_reader_dialog
0.45s call     napari/_qt/menus/_tests/test_file_menu.py::test_show_shortcuts_actions
0.21s setup    napari/_qt/menus/_tests/test_file_menu.py::test_sample_data_triggers_reader_dialog
0.14s teardown napari/_qt/menus/_tests/test_file_menu.py::test_sample_data_triggers_reader_dialog
0.04s setup    napari/_qt/menus/_tests/test_file_menu.py::test_show_shortcuts_actions
0.03s teardown napari/_qt/menus/_tests/test_file_menu.py::test_show_shortcuts_actions
============================================================= 2 passed in 2.23s ==============================================================
```

**Review by DragaDoncila — APPROVED** [review]
After some more investigation it turns out the shortcut works fine, *except* when there's no layers open and the canvas is in focus i.e. if there's any layers open, or you just click somewhere on the window but not on the canvas, it works fine. I suspect this is something to do with the welcome overlay.

Since this also works fine for Mac, and @Czaki confirmed he's not seeing this issue on Ubuntu 21.10, I'm approving here and will raise a separate issue for the specific welcome overlay issue.

File: napari/utils/interactions.py (hunk 1 above)
Comment (Carreau) [reply napari/utils/interactions.py:239]: We use `-` in the definition of shortcuts almost everywhere, and I don't think endusers are exposed to QT internals. As this is used to display shortcuts I believe, I want to make sure it's consistent UI wise.

**Comment by Carreau** [comment]
Do you know why it was removed ? Could it be that it conflict with something else on some keyboard layouts ?

**Review by Carreau — APPROVED** [review]
No objections, just some questions.

File: napari/utils/interactions.py (hunk 1 above)
Comment (Czaki) [reply napari/utils/interactions.py:239]: > We use `-` in the definition of shortcuts almost everywhere, and I don't think endusers are exposed to QT internals. As this is used to display shortcuts I believe, I want to make sure it's consistent UI wise.

Here you have a screenshot of the file menu where all shortcuts are rendered by Qt using its representation:

Zrzut ekranu z 2022-06-09 10-26-45

Also on welcome screen we use Qt convention

Zrzut ekranu z 2022-06-09 10-29-04

And the only place that I found that users see napari shortcuts is Shortcut part of the Preferences Window. 

Zrzut ekranu z 2022-06-09 10-29-45

**Comment by Czaki** [comment]
> Do you know why it was removed ? Could it be that it conflict with something else on some keyboard layouts ?

As I check on 0.4.10 release (last with working shortcut) it was opening separate windows with information about shortcuts. In 0.4.11 we have the option to edit shortcuts in preferences. So it looks like someone replaces one interface with another and forget to implement shortcuts.

**Comment by Carreau** [comment]
Ok, fair enough, let's get that in then.

**Pull request merged**